test/time_test.py

```diff
@@ -19,30 +28,35 @@ def test_get_fps(self):
         c = Clock()
         self.assertEqual(c.get_fps(), 0)
         # Type check get_fps should return float
-        self.assertTrue(type(c.get_fps()) == float)
-        # Allowable margin of error in percentage
-        delta = 0.30
-        # Test fps correctness for 100, 60 and 30 fps
-        self._fps_test(c, 100, delta)
-        self._fps_test(c, 60, delta)
-        self._fps_test(c, 30, delta)
-
-    def _fps_test(self, clock, fps, delta):
+        self.assertTrue(isinstance(c.get_fps(), float))
+
+        # Test fps correctness for a few fps
+        for i in [24, 30, 60, 100, 120]:
+            self._fps_test(c, i)
+
+    def _fps_test(self, clock, fps, reccnt=0):
         """ticks fps times each second, hence get_fps() should return fps"""
+        if reccnt == NUM_TRIES:
+            self.fail("get_fps does not work")
+            
+        delta = 0.2
         delay_per_frame = 1.0/fps
-        for f in range(fps):  # For one second tick and sleep
+        for f in range(fps // 4):  # For 250ms tick and sleep
             clock.tick()
             time.sleep(delay_per_frame)
+        
         # We should get around fps (+- fps*delta -- delta % of fps)
-        self.assertAlmostEqual(clock.get_fps(), fps, delta=fps*delta)
+        if not _is_almost_equal(clock.get_fps(), fps, delta):
+            # recurse on failure
+            self._fps_test(clock, fps, reccnt + 1)
 
     def test_get_rawtime(self):
-
-        iterations = 10
-        delay = 0.1
-        delay_miliseconds = delay*(10**3) #actual time difference between ticks
-        framerate_limit = 5
-        delta = 50 #allowable error in milliseconds
+        iterations = 5
+        delay = 0.02
+        delay_miliseconds = delay * 1000
```

Comment:
typo here in word 'miliseconds' - looks like it is an old typo that has just been preserved. Should be 'milliseconds' with two L's.

Reply:
Done :)